src/github.com/matrix-org/dendrite/roomserver/input/consumer.go

```diff
@@ -45,6 +46,11 @@ type Consumer struct {
 	// The ErrorLogger for this consumer.
 	// If left as nil then the consumer will panic when it encounters an error
 	ErrorLogger ErrorLogger
+	// If non-nil then the consumer will stop processing messages after this
+	// many messages and will shutdown
+	StopProcessingAfter *int
```

Comment:
Be more precise here please. It's unclear if a malformed message counts as having been "processed" or not. Something like:

> If non-nil then the consumer will stop processing messages after the given number of messages and will shut down. Malformed messages are included in this count.